### PR TITLE
Fix for windows without a WM_CLASS

### PR DESCRIPTION
**Pull request by beachmachine**
Ignores windows without a WM_CLASS and also sets the theme when the WM_CLASS changes. This makes the new Spotify version work again.

### CODE REVIEW

**Comment by felipegarcia92** [comment]
@oktayacikalin first, thanks for writing this, man. 

secondly, maybe you can get this merged?

File: switcher.py [hunk 1]
```diff
@@ -58,16 +70,20 @@ def on_settings_changed(self, settings, keys):
...
         screen = Wnck.Screen.get_default()
         for window in screen.get_windows():
-            instance_name = window.get_class_instance_name().lower()
+            instance_name = window.get_class_instance_name()
+            if instance_name is None:
+                continue
+            instance_name = instance_name.lower()
             if instance_name in changes:
-                self.update_window(screen, window)
+                window.connect('class-changed', self.on_window_class_changed)
```
Comment (oktayacikalin) [inline switcher.py]: I'm not sure but does this overwrite the existing binding? Otherwise this would be better put elsewhere.

**Comment by oktayacikalin** [comment]
As you might have guessed, I've not used this tool for some time. Reason is that I'm using a Chromebook as my daily desktop driver.

It would be really great if you or someone else could verify and eventually fix the problem I commented above.

**Comment by Venemo** [comment]
I can confirm that this pull request fixes Spotify. Will you merge it?

**Comment by vk2r** [comment]
It still does not work with the latest version of Spotify.

**Comment by Tomires** [comment]
Can confirm this works for Spotify (as of Nov '17). Please merge!
screenshot from 2017-11-19 14 13 23

**Pull request merged**